server/request_handler.go

```diff
@@ -134,11 +134,11 @@ func (r *RpcRequestHandler) process() {
 	r.logger = r.logger.New("rpc_method", jsonReq.Method)
 
 	// Process single request
-	r.processRequest(client, jsonReq, origin, referer, isWhitehatBundleCollection, whitehatBundleId, urlParams, r.req.URL.String())
```

Comment:
I wonder if a better approach would be not to pass the full body around for possible later signature check, but instead here do a signature check if the header is present, and only pass the result around.

Downside of that approach: doing the signature check also on requests that wouldn't need it. Upside: everything is a bit simpler.

Reply:
Ya I like that too, I'll make that change and add tests for the signatures.

Reply:
Refactored to that approach in https://github.com/flashbots/rpc-endpoint/pull/151/commits/d90fb9b1c8de38358dbf4e5c748e52b234462300.